packages/http-specs/specs/payload/pageable/mockapi.ts

```diff
@@ -38,12 +44,243 @@ Scenarios.Payload_Pageable_ServerDrivenPagination_link = passOnSuccess([
     response: {
       status: 200,
       body: json({
-        pets: [
-          { id: "3", name: "bird" },
-          { id: "4", name: "fish" },
-        ],
+        pets: SecondPage,
       }),
     },
     kind: "MockApiDefinition",
   },
 ]);
+
+function reqBodyResBodyHandler(req: MockRequest) {
+  switch (req.body?.token) {
+    case undefined:
+      return {
+        status: 200,
+        body: json({
+          pets: FirstPage,
+          nextToken: "12345",
+        }),
+      } as const;
+    case "12345":
+      return {
+        status: 200,
+        body: json({
+          pets: SecondPage,
+        }),
+      } as const;
+    default:
+      throw new ValidationError(
+        "Unsupported continuation token",
+        `"undefined" | "12345"`,
+        req.body.token,
+      );
+  }
+}
+
+Scenarios.Payload_Pageable_ServerDrivenPagination_ContinuationToken_reqBodyResBody = passOnSuccess([
+  {
+    uri: "/payload/pageable/server-driven-pagination/continuationtoken/request-body-response-body",
+    method: "get",
+    request: {},
+    response: EmptyResponse,
```

Comment:
just curious why is empty response?

Reply:
No use but will report build error if set with `response: {}` ...

Reply:
Why set with `response: {}`? This is not the expected response.

Reply:
I mean I don't want to waste time to repeat the response content again. And you can see the expected response in handler part (e.g. [here](https://github.com/microsoft/typespec/blob/ba4c0bb2031900a7aaecda093ddfe1f4edfd16cf/packages/http-specs/specs/payload/pageable/mockapi.ts#L91-L115))

Reply:
What I take, is these cases would also serve other purpose like server-side tests. Therefore, the definition of mockApi  should be as close to what is valid (where the "handler" part is a solution that only works for client-orientated test).

I will leave this to Timothee to decide.

Reply:
I see. I think it makes sense to add some code for server side test.